internal/safe/safe.go

```diff
@@ -57,7 +58,8 @@ type id interface {
 }
 
 func ID(id id) string {
-	if id == nil {
+	v := reflect.ValueOf(id)
```

Comment:
Isn't it overkill? How much does it cost for us when such a callback is called many times?